### DISCUSSION TITLE
External Logstash to Security Onion

### DISCUSSION THREAD

**Discussion by ColeVan** [Security-Onion-Solutions/securityonion#8685] [Unsupported Versions]
Hello everyone!

Question? What's the best way to send logs from a external Logstash to Security Onion?

**Comment by reyesj2** [comment]
Your setup would look something like if you're running distributed
Output pipeline from external logstash to manager node -> new input pipeline on security onion manager -> redis -> input/output pipeline on search node to Elasticsearch

**Reply by ColeVan** [reply]
Hello. Have you done this successfully before? Do you have example pipeline conf's you can share?

**Reply by reyesj2** [reply]
I have not, the default pipelines on Security Onion are located in **/opt/so/saltstack/default/salt/logstash/pipelines/config/so**

**Reply by ColeVan** [reply]
After following Elastic documentation I found that using lumberjack plugin is the way to go if you want to send data from external Logstash to SO2 logstash.

However, the data needs to be sent from external Logstash instance securely using ssl certs to SO2 manager over port 5044. What would be the best way forward? Anyone have example logstash pipeline configurations or suggestions on how to accomplish this?

**Comment by weslambert** [comment]
Like you mentioned, the best approach here would be to use the [Lumberjack output plugin ](https://www.elastic.co/guide/en/logstash/current/plugins-outputs-lumberjack.html)from the external instance to send to Security Onion, where you would configure the [Lumberjack input plugin](https://www.elastic.co/guide/en/logstash/current/plugins-inputs-lumberjack.html).

You should be able to configure encryption similar to the following:
https://docs.securityonion.net/en/2.3/beats.html?highlight=beats#using-the-beats-default-port-5044-with-encryption

**Reply by ColeVan** [reply]
Thanks for getting back to me Wes. 

When installing the lumberjack plug-in on airgapped SO2 managers do you forsee any complications or problems I might encounter?

**Reply by ColeVan** [reply]
For plugins not bundled by default running bin/logstash-plugin install logstash-input-lumberjack should work, but is that done inside of SO2 manager logstash docker?

**Reply by ColeVan** [reply]
Looking into more documents suggest creating certs on the downstream logstash instance for the upstream logstash instance to communicate over. I wonder how this would effect SO2 managers with private IP's behind a firewall??

https://www.claudiokuenzler.com/blog/963/logstash-to-logstash-logging-forwarding-lumberjack-output-plugin

**Reply by ColeVan** [reply]
Mild success! Using the link above I was able to successfully ingest logs from External Logstash to SO2 Manager along with proper ssl cert setup
![image](https://user-images.githubusercontent.com/70167373/189451758-6fc2f596-6a43-4f1b-b067-928d4d99a252.png)
![image](https://user-images.githubusercontent.com/70167373/189451762-7ad4e1a9-9ec5-4e13-b14d-10c119eca65c.png)
![image](https://user-images.githubusercontent.com/70167373/189451769-fd02d30b-6d02-4974-bb5a-2013639a6f6b.png)

**Reply by ColeVan** [reply]
I finally put everything into production. I'm ingesting logs from external Logstash to SO2 manager flawlessly. The setup was rather simple after figuring it out. Be sure to place codec=>json in your input conf on SO2 manager so that the data comes in parsed. I'm running SO2.3.140 and I'm able to parse logs from winlogbeats version 7.10 and above that are sent to external Logstash and shipped to mine.

**Reply by weslambert** [reply]
Good to hear! Although, I think we do not current bundle the lumberjack input, IIRC. In this case, you could still use something like the TCP or UDP input for something with a bit more persistence.
